libnavigation-ui/src/main/java/com/mapbox/navigation/ui/camera/NavigationCamera.java

```diff
@@ -0,0 +1,580 @@
+   * last a specified amount of time given in milliseconds. A callback can be used to be notified when animating
+   * the camera stops. During the animation, a call to {@link MapboxMap#getCameraPosition()} returns an intermediate
+   * location of the camera in flight. If the camera is in a tracking mode,
+   * this animation is going to be ignored, or break the tracking, based on the {@link CameraUpdateMode}.
+   *
+   * @param update     the change that should be applied to the camera.
+   * @param durationMs the duration of the animation in milliseconds. This must be strictly
+   *                   positive, otherwise an IllegalArgumentException will be thrown.
+   * @param callback   an optional callback to be notified from the main thread when the animation
+   *                   stops. If the animation stops due to its natural completion, the callback
+   *                   will be notified with onFinish(). If the animation stops due to interruption
+   *                   by a later camera movement or a user gesture, onCancel() will be called.
+   *                   Do not update or animate the camera from within onCancel(). If a callback
+   *                   isn't required, leave it as null.
+   * @see CameraUpdateMode for how this update interacts with the current tracking
+   */
+  public void update(NavigationCameraUpdate update, int durationMs, @Nullable MapboxMap.CancelableCallback callback) {
+    animationDelegate.render(update, durationMs, callback);
+  }
+
+  /**
+   * Call in {@link FragmentActivity#onStart()} to properly add the {@link RouteProgressObserver}
+   * and {@link LocationObserver}
+   * for the camera and prevent any leaks or further updates.
+   */
+  @OnLifecycleEvent(Lifecycle.Event.ON_START)
+  public void onStart() {
+    if (navigation != null) {
+      navigation.registerRouteProgressObserver(routeProgressObserver);
+      navigation.registerLocationObserver(locationObserver);
+    }
+  }
+
+  /**
+   * Call in {@link FragmentActivity#onStop()} to properly remove the
+   * {@link RouteProgressObserver} and {@link LocationObserver}
+   * for the camera and prevent any leaks or further updates.
+   */
+  @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+  public void onStop() {
+    if (navigation != null) {
+      navigation.unregisterRouteProgressObserver(routeProgressObserver);
+      navigation.unregisterLocationObserver(locationObserver);
+    }
+  }
+
+  /**
+   * This method can be called if you did not pass an instance of {@link MapboxNavigation}
+   * to the constructor.
+   * <p>
+   * The camera will begin listening to progress updates and update the route accordingly.
+   *
+   * @param navigation to add the camera progress change listener
+   */
+  public void addProgressChangeListener(MapboxNavigation navigation) {
+    this.navigation = navigation;
+//    navigation.setCameraEngine(new DynamicCamera(mapboxMap));
```

Comment:
Why we commented out this?

Reply:
`Camera` doesn't live in `MapboxNavigation` anymore.

Reply:
Now that is going to live under `libnavigation-ui` we should revisit `Camera` implementation. Let's cut a ticket to track this work.